### PR TITLE
Upstream PR #339

### LINKED ISSUE #314: Docker-dev is missing killall command
**Describe the bug**
The test in demo/chaincode/fpc requires `killall` command to kill the mock server after tests are finished. Our docker dev-image misses that command.

**To Reproduce**
```
cd utiles/docker
make dev; make run
cd demo/chaincode/fpc
make
```

**Expected behavior**
```
Test successful.
root@7984cc9a34dc:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/demo/chaincode/fpc#
```

**Log-filesScreenshots**
```
Test successful.
./test.sh: line 11: killall: command not found
Makefile:48: recipe for target 'test' failed
make: *** [test] Error 127
root@7984cc9a34dc:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/demo/chaincode/fpc#
```

**Environment (please complete the following information):**
MacOS using docker-dev images from master (42b34e4)

**Additional context**

### PR DESCRIPTION
**Pull request by g2flyer**
This is an upstreamed version of PR #339 so we have docker image versioning, a robust end-to-end workable dev(elopment container) and a few other reliability and docker build improvements also in master

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #314 and #346.  makes a big step towards #348; provides short-term fix for #269.

**Special notes for your reviewer**:

There are a few small changes compared to PR #339 to adopt to changes in master plus an additional version consistency issue (last commit).  If you want to do commit-by-commit comparison, not that the merge of #339 unfortunately squashed all commits, so you won't see the commits when checking out `concept-release-1.0`.  However, you still will have the commits in the commit history from #339 
Regarding testing, it is essentially the same as for #339, although now lighter as there is no demo & fpc docker networks to test (and hence build and travis time will also be faster)

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -74,10 +74,15 @@ before_install:
...
 
 script:
-  # travis doesn't check out the branch itself but for our make to work we
-  # need a branch or alike. So check create a local one ...
-  - git checkout -b we-need-a-branch-for-below-to-work
-  - make
+  # * test directly on host
+  # - make all clobber
+  # NOTE: disabled as it will reach the 50 min limits. Testing inside dev container
+  # hopefully/presumably should hopefully be a super-set of the host build/test
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply .travis.yml:83]: we could do this also in another PR

**Comment by mbrandenburger** [comment]
Apparently travis also has the same docker problems as you can see here in the logs although FPC build runs through.

https://travis-ci.com/github/hyperledger-labs/fabric-private-chaincode/builds/164589085#L13176

**Comment by g2flyer** [comment]
> When running `make` inside the dev container, I can see some docker-related errors but make continues ... not really sure what the problem is. See log

Bruno also observed this during the review of the cr1.0.1 version (and linked to a related docker issue).  However, i've seen that problem locally since a long time and i think it even existed for travis.  Luckily, though, except for polluting the log-files, it doesn't seem to cause any other problems ....

File: .travis.yml [hunk 2]
```diff
@@ -77,6 +77,9 @@ before_install:
...
 
 script:
+  # - test directly on host
   - make all clobber
-  # Note: travis of couse does an implicit clobber but adding it here makes sure
-  # it works properly
+  #   Note: travis of couse does an implicit clobber but adding it here makes sure
+  #   it works properly, also to make sure below is clean-slate
+  # - test that it also runs properly in dev container  
+  - make -C utils/docker run DOCKER_DEV_OPTIONAL_CMD='make'
```
[the remark this answers is not in the document]
Comment (g2flyer) [reply .travis.yml]: parallelizing should somewhat help but also note given the implicit docker sharing, you don't get as much speed-up as you would might expect.  Nevertheless, doing your proposed change makes sense (but i propose you push it afterwards as a separate PR?)

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (g2flyer) [reply .travis.yml:83]: Good point that when using docker to build, then we don't really need the host installation anymore (and so also no duplicate installation steps in addition to what we already have in docker and in docu).  Depends of course also what we do with your separate parallelization proposal: Maybe easiest is to just leave it as is but open a github issue capturing the discussion here?

[the remark this answers is not in the document]
Comment (mbrandenburger) [reply .travis.yml:83]: ok - we will tackle this in another PR

**Comment by mbrandenburger** [comment]
Thanks!

**Pull request merged**